given/EventEnvelopeExtensions.cs

```diff
@@ -0,0 +1,19 @@
+using Dolittle.Time;
+
+namespace Dolittle.Runtime.Events.Store.Specs.given
+{
+    public static class EventMetadataExtensions
+    {
+        public static bool LossyEquals(this EventMetadata first, EventMetadata second)
```

Comment:
Just move this into EventMetadata and override Equals (you can remove the Value<EventMetadata> inheritance since you are providing the equals method yourself).